app/main.py

```diff
@@ -1,3 +1,18 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    cached = {}
+
+    @wraps(func)
+    def inner(*args) -> Callable:
+        if args in cached:
+            print("Getting from cache")
+            return cached[args]
+        else:
+            result = func(*args)
+            cached[args] = result
+            print("Calculating new result")
+            return result
```

Comment:
Try to rewrite your decorator not to have `return` duplicated

Reply:
done